lib/rpc/handler/FallbackHandler.ts

```diff
@@ -0,0 +1,97 @@
+import Logger from 'bunyan'
+import { ChainId } from '@uniswap/sdk-core'
+import { getProviderId } from '../utils'
+import { ProviderHealthStateRepository } from '../ProviderHealthStateRepository'
+import { ProviderHealthStateDynamoDbRepository } from '../ProviderHealthStateDynamoDbRepository'
+import { ProviderHealthState } from '../ProviderHealthState'
+import { MetricLoggerUnit } from '@uniswap/smart-order-router'
+import { metricScope, MetricsLogger } from 'aws-embedded-metrics'
+import { APIGatewayProxyResult } from 'aws-lambda'
+import { AWSMetricsLogger } from '../../handlers/router-entities/aws-metrics-logger'
+
+interface AlarmEvent {
+  alarmName: string
+  state: string
+  previousState: string
+  providerId: string
+  reason: string
+}
+
+export class FallbackHandler {
+  private healthStateRepository: ProviderHealthStateRepository
+  private log: Logger
+
+  constructor(log: Logger) {
+    const dbTableName = process.env.PROVIDER_HEALTH_STATE_DB_TABLE_NAME
+    if (!dbTableName) {
+      throw new Error('Missing DB_TABLE_NAME in env var')
+    }
+    this.healthStateRepository = new ProviderHealthStateDynamoDbRepository(dbTableName, log)
+    this.log = log
+  }
+
+  private buildHandler() {
+    return metricScope((metricsLogger: MetricsLogger) => async (event: object): Promise<APIGatewayProxyResult> => {
+      metricsLogger.setNamespace('Uniswap')
+      metricsLogger.setDimensions({ Service: 'RoutingAPI' })
+      const metric = new AWSMetricsLogger(metricsLogger)
+
+      const alarmEvent = this.readAlarmEvent(event)
+      this.log.debug({ alarmEvent }, 'Parsed alarmEvent')
+
+      if (
+        (alarmEvent.previousState === 'OK' || alarmEvent.previousState === 'INSUFFICIENT_DATA') &&
```

Comment:
not sure if the previous state really matters: what do we lose if we only write `if(alarmEvent.state === 'ALARM'` ?

Reply:
I only want to take action when the state actually changes. This serves as a guard.

Also if you consider the "recover" case, we do need to consider previous state. For example, we don't want to update the DB state when states changes from "OK" to "INSUFFICIENT_DATA", but we do when states changes from "ALARM" to "INSUFFICIENT_DATA".

Reply:
Addressed in other comments. Code logic is fixed as suggested.